### PR TITLE
Only return the casting button when there are devices to cast to.

### LINKED ISSUE #179: Hide "Cast" button when there are no devices to cast
## Feature description

I kindly suggest hiding the "Cast" button when there are no devices to cast. We currently display an empty button and nothing happens when this button is clicked.

image (40)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by dero**
## Summary

Only resolve the cast button promise when cast receiver devices are detected.

Fixes #179

### CODE REVIEW

**Review by beaufortfrancois — CHANGES_REQUESTED** [review]
The casting button would not be removed when cast devices are not available anymore sadly.

I believe we should have a `displayCastState()` function that would simply toggle cast button visibility when `CAST_STATE_CHANGED` event is fired. WDYT?

**Comment by dero** [comment]
@beaufortfrancois Right. I wasn't sure if dynamically removing the cast button from the once established UI without indicating what's happened would be a good solution from the UX perspective.

But I guess with casting it's a relatively common pattern for the button to only appear when cast targets are available.

Updated in 7cdfcdd. Would that work?

File: src/index.js [hunk 1]
```diff
@@ -170,16 +170,32 @@ window.kinoInitGoogleCast = (function kinoInitGoogleCastIIFE() {
   return () => {
     if (!castButtonPromise) {
       castButtonPromise = new Promise((resolve) => {
+        const castButton = document.createElement('button');
+        const castCustomElement = document.createElement('google-cast-launcher');
+
+        castButton.setAttribute('aria-label', 'Cast this video');
+        castButton.appendChild(castCustomElement);
+        castButton.classList.add(CAST_BUTTON_HIDDEN_CLASSNAME);
```
Comment (beaufortfrancois) [inline src/index.js:178]: This line is not needed as `castButton` will only be added to the document once promise is resolved right?

Comment (dero) [reply src/index.js:178]: @beaufortfrancois The button is added to the document as soon as the Google Cast API becomes available after the recent changes – regardless of whether cast devices are available or not.

The button visibility (controlled using the class name) then depends on cast state changes.

Comment (dero) [reply src/index.js]: Quite right. 🤦‍♂️

Thank you. Removed in 3cc1ef6.

**Review by beaufortfrancois — APPROVED** [review]
Thank **you** for working on this!

**Comment by felipebochehin87** [comment]
Looks good to me. Cast button is not displayed when there are no devices to cast to.

no cast button

**Pull request merged**